fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+            # ids are the encoded tokens,
+            # type_ids are for things like "which sentence is this from". There seem to be several limitations on those:
+            #   - they must be unique, i.e. several different sentences cannot receive the same type_id from outside (for some reason they can be mapped
+            #     to the same value if wrong ids were supplied by the user
+            #   - it seems like the ids must be consecutive, starting with 0, in order to work as expected. If they do not start with 0,
+            #     it is forced on the last sequence.
+            # tokens are the actual tokens (for example - ['c1ccc(', '/C(', '=N/N', 'c2nc3ccccc3', 's2)', 'c2cccc', 'n2)cc1', '[PAD]', '[PAD]', '[PAD]'])
+            # offsets describe the starting point and length of each original token
+            # attention_mask - by default puts 1 for everything that isn't padding, and 0 for those that are padding
+            # special_tokens_mask - 1 for anything that is a special token (e.g. padding, separator, etc.) 0 for the rest
+            # overflowing - It's a list of Encoding structures of original content that got clipped out due to max length definition.
+            #               In my experience, only the zeroth index contains anything. Don't know when there's more then one member in the list.
+
+        merged_encoding = Encoding.merge(encoded_list)
+
+        if max_len is not None:
+            merged_encoding.truncate(max_length=max_len)
+        if padding_token_id is not None:
+            # TODO: find the actual padding token ID
+            rnd_type = list(self.tokenizers_info.keys())[0]
+            rnd_inst = self.tokenizers_info[rnd_type]["tokenizer_inst"]
+            padding_token_id = rnd_inst.token_to_id(padding_token)
+            merged_encoding.pad(
+                length=max_len,
+                direction="right",
+                pad_id=padding_token_id,
+                pad_token=padding_token,
+                pad_type_id=pad_type_id,
+            )
+
+        return merged_encoding
+
+    def decode(self, ids: Iterable, skip_special_tokens: Optional[bool] = False) -> str:
+        """Receives a list of IDs and returns a string of tokens
+            TODO: possibly output also the type of token (AA, SMILES, etc)
+        Args:
+            ids (Iterable): _description_
+            skip_special_tokens (Optional[bool], optional): _description_. Defaults to False.
+
+        Returns:
+            str: _description_
+        """
+
+        if skip_special_tokens:
+            ret_val = [self.decoder_dict[id]["token"] for id in ids if not self.decoder_dict[id]["is_special"]]
+        else:
+            ret_val = [self.decoder_dict[id]["token"] for id in ids]
+        return "".join(ret_val)
+
+    def get_token_id(self, token: str) -> int:
+        raise Exception("not implemented yet")
```

Comment:
this one is important to implement

Reply:
token_to_id method does this. Removed the declaration.